### PR TITLE
feat(goal_planner):  dense goal candidate sampling in BusStopArea

### PR DESCRIPTION
**Pull request by soblin**
## Description

depends https://github.com/autowarefoundation/autoware.universe/pull/8794

launcher PR: https://github.com/autowarefoundation/autoware_launch/pull/1156

## Related links

- [CompanyName internal link](https://tier4.atlassian.net/browse/RT1-4047)

### use_bus_stop_area = false

image

### use_bus_stop_area = true

image

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp [hunk 1]
```diff
@@ -165,12 +177,18 @@ GoalCandidates GoalSearcher::search(const std::shared_ptr<const PlannerData> & p
       const auto transformed_vehicle_footprint =
         transformVector(vehicle_footprint_, autoware::universe_utils::pose2transform(search_pose));
 
-      if (isInAreas(transformed_vehicle_footprint, getNoParkingAreaPolygons(pull_over_lanes))) {
+      if (
+        parameters_.bus_stop_area.use_bus_stop_area &&
+        !isInAreas(transformed_vehicle_footprint, bus_stop_area_polygons)) {
+        break;
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp:183]:
```suggestion
        continue;
```

Comment (kosuke55) [reply planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp:183]: sorry the name of `isInAreas` is bad, it return true, if it intersects.
for bus stop are the footprint should be completely in the area

**Comment by soblin** [comment]
@kosuke55 Can you approve this ?

**Review by kosuke55 — APPROVED** [review]
LGTM

**Pull request merged**